rate_limiter_channel.go

```diff
@@ -0,0 +1,29 @@
+package tailor
+
+import (
+	"time"
+)
+
+type ChannelBasedRateLimiter struct {
+	t *time.Ticker
+}
+
+// NewChannelBasedRateLimiter creates an instance of rate limiter, which ticker ticks every period to limit the lps.
+func NewChannelBasedRateLimiter(lps int) *ChannelBasedRateLimiter {
+	return &ChannelBasedRateLimiter{
+		t: time.NewTicker(1 / (time.Duration(lps) * time.Second)),
+	}
+}
+
+func (rl *ChannelBasedRateLimiter) Allow() bool {
+	select {
```

Comment:
Can be 
```go
_, ok := <-rl.t.C
return ok
```

Reply:
No one closes it.

Reply:
Except strange users which Close manually()